src/modules/analytics/CommonWords.tsx

```diff
@@ -0,0 +1,142 @@
+  );
+
+  const deleteCustomWord = (word: string): void => {
+    setSelectedOtherWords(selectedOtherWords.filter((w) => w !== word));
+  };
+  const selectCommonChip = (text: string): void => {
+    setSelectedCommonWords([...new Set([...selectedCommonWords, text])]);
+  };
+
+  return (
+    <Stack spacing={2} mt={2}>
+      <Typography variant="h6">{t('MOST_FREQUENT_WORDS_TITLE')}</Typography>
+      <Typography variant="body1">{t('FILTER_BY_COMMON_KEYWORDS')}</Typography>
+      <Stack spacing={1} direction="row">
+        {Object.entries(mostFrequentWords).map(([text, count]) => (
+          <KeywordChip
+            key={text}
+            text={text}
+            count={count}
+            isSelected={selectedCommonWords.indexOf(text) > -1}
+            onClick={() => selectCommonChip(text)}
+          />
+        ))}
+        <Button
+          onClick={() => setSelectedCommonWords(topRepetitiveWords)}
+          variant={isAllSelected ? 'contained' : 'outlined'}
+        >
+          {t('ALL')}
+        </Button>
+      </Stack>
+      <Stack spacing={1} sx={{ maxWidth: 300 }}>
+        <Typography variant="body1">{t('SEARCH_BY_OTHER_KEYWORDS')}</Typography>
+
+        <TextField
+          size="small"
+          onKeyDown={(event) => {
+            if (event.key === 'Enter') {
+              setSelectedOtherWords([
+                ...new Set([...selectedOtherWords, otherWord]),
+              ]);
+              setOtherWord('');
+            }
+          }}
+          value={otherWord}
+          onChange={(e) => {
+            setOtherWord(e.target.value);
+          }}
+          placeholder={t('SEARCH_COMMON_WORDS_PLACEHOLDER')}
+        />
+        <Stack spacing={1} direction="row">
+          {selectedOtherWords.map((text) => (
+            <Chip
+              key={text}
+              label={text}
+              variant="outlined"
+              onDelete={() => deleteCustomWord(text)}
+            />
+          ))}
+        </Stack>
+      </Stack>
+      <Grid container sx={{ height: '500px' }}>
+        <Grid item xs={12} md={6} sx={{ height: '100%', overflowY: 'auto' }}>
+          <Stack spacing={2} p={1}>
+            {commentsMatchSelectedWords.map((ele) => (
+              <TextWithHighlightedKeywords
+                key={ele.id}
+                sentence={ele.data.content}
+                memberName={ele.member.name}
+                words={[...selectedCommonWords, ...selectedOtherWords]}
+                onClick={() => setChatMemberID(ele.member.id)}
+              />
+            ))}
+
+            {commentsMatchSelectedWords.length === 0 && (
+              <Typography mt={2}>{t('NO_RESULTS_MATCH_WORDS')}</Typography>
+            )}
+          </Stack>
+        </Grid>
+        {chatMemberID && (
+          <Grid item xs={12} md={6} sx={{ height: '100%', overflowY: 'auto' }}>
+            <PlayerView id={chatMemberID} />
+          </Grid>
+        )}
+      </Grid>
+    </Stack>
+  );
+};
```

Comment:
In general this was difficult to understand because your nomenclature isn't clear. Nevertheless I believe you could merge the `selectedFrequentWords` with `selectedCustomWords` if you define `frequentWords` as prop. The `frequentWords` will be used to be always shown as chips, and also the default selected words. While clicking on any chip or typing words, all of them go to `selectedWords`. The set of `selectedWords`+`frequentWords` can then be used to show chips, and `selectedWords` only can be used to define whether the chips are selected. Would that make sense?